website/docs/concepts/contexts.mdx

```diff
@@ -96,35 +96,57 @@ A context provider is required to consume the context. `ContextProvider<T>`, whe
 The children are re-rendered when the context changes. A struct is used to define what data is to be passed. The `ContextProvider` can be used as:
 
 ```rust
-use std::rc::Rc;
 use yew::prelude::*;
 
+
+/// App theme
 #[derive(Clone, Debug, PartialEq)]
 struct Theme {
     foreground: String,
     background: String,
 }
 
+/// Main component
 #[function_component]
-fn NavButton() -> Html {
-    let theme = use_context::<Theme>();
+pub fn App() -> Html {
+    let ctx = use_state(|| Theme {
+        foreground: "#000000".to_owned(),
+        background: "#eeeeee".to_owned(),
+    });
 
     html! {
-        // use theme
+        // `ctx` is type `Rc<UseStateHandle<Theme>>` while we need `Theme`
+        // so we deref it.
+        // It derefs to `&Theme`, hence the clone
+        <ContextProvider<Theme> context={(*ctx).clone()}>
+            // Every child here and their children will have access to this context.
+            <Toolbar />
+        </ContextProvider<Theme>>
     }
 }
 
+/// The toolbar.
+/// This component has access to the context
```

Comment:
Either it's the first sentence (one line) or it's a new paragraph, in which case we need spacing

```suggestion
/// The toolbar.
///
/// This component has access to the context.
```

```suggestion
/// The toolbar. This component has access to the context.
```